prow/config/jobs.go

```diff
@@ -238,9 +246,12 @@ func (br Brancher) Intersects(other Brancher) bool {
 }
 
 // RunsAgainstChanges returns true if any of the changed input paths match the run_if_changed regex.
-func (ps Presubmit) RunsAgainstChanges(changes []string) bool {
+func (cm ChangeMatcher) RunsAgainstChanges(changes []string) bool {
+	if cm.RunIfChanged == "" {
```

Comment:
Why didn't this check need to exist before?

Reply:
It seems the check was already being made everywhere `RunsAgainstChanges` was used. `reChanges` [isn't always set](https://github.com/kubernetes/test-infra/blob/3c763287157e7040f51d3d84fec2e1e433e9d4df/prow/config/config.go#L1172), creating the risk of nil deref, so I figured it'd be safer to put the check in the function here.